client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+          {tabs.map(tab =>
+            selectedTabId === tab.id ? (
+              <input
+                aria-label="Sheet Name"
+                className={classnames(
+                  Classes.INPUT,
+                  errors.name && Classes.INTENT_DANGER
+                )}
+                key={tab.id}
+                name="name"
+                readOnly={isSubmitting}
+                ref={register({
+                  required: true,
+                })}
+              />
+            ) : (
+              <Tab disabled id={tab.id} key={tab.id}>
+                {tab.name}
+              </Tab>
+            )
+          )}
+          <Tabs.Expander />
+          <Button icon="add" minimal disabled>
+            Add Sheet
+          </Button>
+        </Tabs>
+      ) : (
+        // When the above tab bar isn't rendered, still render a hidden input to house the sheet
+        // name. Rendering no input at all interferes with react-hook-form's state management,
+        // specifically its isDirty tracking
+        <input name="name" ref={register()} style={{ display: 'none' }} />
+      )}
+
+      <div
+        // Since we have to render two tab bars, we aren't able to use the tab bar's built-in
+        // `panel` prop and have to manually apply attributes for tab panel accessibility
+        aria-labelledby={`${Classes.TAB_PANEL}_${batch.name}_${selectedTabId}`}
+        role="tabpanel"
+      >
+        <BatchResultTallySheetTable>
+          <thead>
+            <tr>
+              <th>Choice</th>
+              <th>Votes</th>
+            </tr>
+          </thead>
+          <tbody>
+            {contest.choices.map(choice => (
+              <tr key={choice.id}>
+                <td>{choice.name}</td>
+                <td>
+                  {isTotalsSheet ? (
+                    <span>
+                      {sum(sheets.map(sheet => sheet.results[choice.id]))}
+                    </span>
+                  ) : (
+                    <>
+                      <input
+                        aria-label={`${choice.name} Votes`}
+                        className={classnames(
+                          Classes.INPUT,
+                          errors.results?.[choice.id] && Classes.INTENT_DANGER
+                        )}
+                        name={`results[${choice.id}]`}
+                        readOnly={isSubmitting}
+                        ref={register({
+                          min: 0,
+                          required: true,
+                          valueAsNumber: true,
+                        })}
+                        type="number"
+                        // Visually hide this input instead of completely unmounting it since
+                        // rendering no input at all interferes with react-hook-form's state
+                        // management, specifically its isDirty tracking
+                        style={{ display: !isEditing ? 'none' : undefined }}
```

Comment:
This seemed weird to me, so I did a little investigation. My understanding is that `react-hook-form` determines `isDirty` by comparing the `defaultValues` to the current state of the form inputs. When we remove the inputs from the DOM, react-hook-form automatically unregisters them, so the current state of the form inputs no longer matches the default values. It looks like there's an option for `useForm` called [`shouldUnregister` ](https://legacy.react-hook-form.com/v6/api#useForm)that you can set to `false` to change this behavior so that the inputs don't get unregistered when removed from the DOM. I played around with it a bit and it seemed to solve the problem!

I'm definitely losing a bit of confidence in this whole uncontrolled inputs approach that react-hook-form takes. I know it's supposed to improve performance, but it's definitely got some tricky bits. I think this would be trivial in a controlled input context.

Reply:
So one funky finding here: After specifying `autoFocus` attributes (https://github.com/votingworks/arlo/pull/1661/commits/744c02c4c289e7879648e2e3233dd70e41c654ad), `formState.isDirty` is once again getting set to `true` when it shouldn't be, specifically for sheets with no tallies entered at the moment I click on another batch.

Removing the `autoFocus` attributes fixes this. 

https://user-images.githubusercontent.com/12616928/199392806-f750bcf8-81fa-4921-819d-7d4f6f421896.mov

To simplify the code, I'm just not going to use `formState.isDirty` and instead open the unsaved changes prompt any time you try to leave a batch while in edit mode.